applications: nrf_desktop: Add BAS documentation (#1666)
NordicPlayground/fw-nrfconnect-nrf: applications/nrf_desktop/doc/bas.rst

GATT Battery Service
####################

> The ``bas`` module is used to define the GATT Battery Service and notify the subscribers about the battery level changes.

Reviewer: ```suggestion
Use the ``bas`` module to enable and define the GATT Battery Service, and notify the subscribers about the battery level changes.
```

Module Events
*************

+---------------------+-------------------------+-------------+---------------+------------------+
| Source Module       | Input Event             | This Module | Output Event  | Sink Module      |
+=====================+=========================+=============+===============+==================+

> | :ref:`battery_meas` | ``battery_level_event`` | ``bas``     |		      |			 |

Reviewer: ```suggestion
| :ref:`battery_meas` | ``battery_level_event`` | ``bas``     |		|		 |
```
Make the vertical lines aligned in the table, please.

Author: On my laptop the lines are aligned, I think its related to the way how Github displays it.

Reviewer 2: these are probably tabs

Author: Should we use spaces instead of tabs in `*.rst` files ?

+---------------------+-------------------------+-------------+---------------+------------------+

Configuration
*************

The module is enabled with the ``CONFIG_DESKTOP_BAS_ENABLE`` option.

> The option is selected by ``CONFIG_DESKTOP_HID_PERIPHERAL`` - Battery Service is required for HID peripheral device.

Reviewer: ```suggestion
The option is selected by ``CONFIG_DESKTOP_HID_PERIPHERAL`` -- Battery Service is required for the HID peripheral device.
```

Implementation details
**********************

The module uses :c:macro:`BT_GATT_SERVICE_DEFINE` to define the GATT Service.

> On ``battery_level_event`` the module updates the battery level and notifies the subscribers.

Reviewer: ```suggestion
On ``battery_level_event``, the module updates the battery level and notifies the subscribers.
```

> More detailed information regarding GATT are available in Zephyr documentation: :ref:`zephyr:bt_gatt`.

Reviewer: ```suggestion
More detailed information about GATT are available in the Zephyr documentation: :ref:`zephyr:bt_gatt`.
```
